### PR TITLE
feat(defaultheader, defaultfooter, defaultcarousel, page.tsx): implem…

### PR DESCRIPTION
**Pull request by trev-org04**
implemented home page

Implemented home page with header, footer, and carousel using TailwindCSS

### CODE REVIEW

File: src/components/DefaultHeader.tsx [hunk 1]
```diff
@@ -3,13 +3,115 @@
 // DefaultHeader component definition.
 
 // NextJS essential imports.
+"use client";
+import { useEffect } from "react";
+import Image from 'next/image';
 
 // Local imports.
+import logo from '../../public/images/logo.png';
```
Comment (KemingHe) [inline src/components/DefaultHeader.tsx:11]: - [ ] Fix needed: don't import/specify path, instead, NextJS auto-recognizes public dir resources. 
- [ ] Read more here: https://nextjs.org/docs/app/building-your-application/optimizing/static-assets

Comment (KemingHe) [reply src/components/DefaultHeader.tsx:11]: TLDR: relative import path bad.

File: src/components/DefaultHeader.tsx [hunk 2]
```diff
@@ -3,13 +3,115 @@
...
+						aria-current="page"
+						href="#"
+						data-twe-nav-link-ref
+						>OSU UREA</a
+					>
+					</li>
+				</ul>
+			
+				<div className="flex items-center">
+					<a
```
Comment (KemingHe) [inline src/components/DefaultHeader.tsx:79]: - [ ] Fix: use NextJS's Link instead of a tag.

File: tailwind.config.js
Comment (KemingHe) [inline tailwind.config.js:1]: I wonder if this can be written as .ts?

Also, what about using and appending to the original tailwind css config?

**Review by KemingHe — CHANGES_REQUESTED** [review]
Awesome job first of all, I would request a bit of refactoring to be more NextJS-native in terms of importing and optimization. See you in the coversation.

**Comment by KemingHe** [comment]
@trev-org04 see your amazing preview deployment here:
https://frontend-preview-app-wrhx7cwynq-uc.a.run.app/

File: tailwind.config.js
Comment (KemingHe) [reply tailwind.config.js:1]: - [ ] @KemingHe will be reverting this change and updating the tailwind routes withe the latest tsconfig setup.

**Comment by trev-org04** [comment]
@KemingHe Pushed changes that address your comments and requesting another review at your earliest convenience

File: src/components/DefaultRegisterForm.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        var lastName = (document.getElementById('lastName') as HTMLInputElement).value;
+        var email = (document.getElementById('email') as HTMLInputElement).value;
+        var password = (document.getElementById('password') as HTMLInputElement).value;
+        var confirmPassword = (document.getElementById('confirmPassword') as HTMLInputElement).value;
+        
+
+        console.log('fName: ' + firstName);
+        console.log('lName: ' + lastName);
+        console.log('email: ' + email);
+        console.log('password: ' + password);
```
[the remark this answers is not in the document]
Comment (KemingHe) [reply src/components/DefaultRegisterForm.tsx]: @trev-org04 
- [ ] Please look into this and resolve the security weakpoint

**Comment by KemingHe** [comment]
@trev-org04 Deployment is broken due to the outdated Firebase config that works in local dev but out compatible with Docker and Cloud Run, will update you soon via my fix on I and Jerry's branch `refactor-onlycomponentsupdate-keminghe`.

**Review by KemingHe — APPROVED** [review]
Fixed everything fixable so far. **I foresee another 2 weeks at least to polish this.** Being realistic.

**Pull request merged**